scikeras/wrappers.py

```diff
@@ -871,192 +874,218 @@ def preprocess_y(y):
 
         Returns:
             y : modified 2D numpy array with 0 indexed integer class labels.
-            extra_args : dictionary of output attributes, ex `n_outputs_`
         """
-        y, extra_args = super(KerasClassifier, KerasClassifier).preprocess_y(y)
-
-        target_type_ = type_of_target(y)
+        y = super().preprocess_y(y, reset=reset)
 
-        if len(y.shape) == 1:
-            n_outputs_ = 1
-        else:
-            n_outputs_ = y.shape[1]
+        loss = self.loss
 
-        if target_type_ == "binary":
+        if self.target_type_ == "binary":
             # y = array([1, 0, 1, 0])
             # single task, single label, binary classification
             model_n_outputs_ = 1  # single sigmoid output expected
+            n_outputs_ = 1
             # encode
-            encoder = LabelEncoder()
-            # No need to reshape to 1D here,
-            # binary targets are always 1D already
+            if reset:
+                encoder = make_pipeline(
+                    Ensure2DTransformer(), OrdinalEncoder(dtype=np.float32),
+                )
+            else:
+                encoder = self.encoders_[0]
```

Comment:
It appears the else branch of the if-else statement doesn't do anything meaningful. This is repeated for a couple other target types.

Reply:
It re-uses the existing encoders, eg if calling `partial_fit`.

Reply:
I realize that I had this:

```python3
if reset:
    encoder = make_pipeline(...)
else:
    encoder = self.encoders_[0]
y = encoder.fit_transform(y)
```

When I should have had:

```python3
if reset:
    encoder = make_pipeline(...)
    y = encoder.fit_transform(y)
else:
    encoder = self.encoders_[0]
    y = encoder.transform(y)
```

That makes the branch used right?

Reply:
Not if you do this:

``` python
if reset:
    encoder = make_pipeline(...).fit(y)
y = encoder.transform(y)
```

Reply:
Where is `encoder` defined if `reset==False`? I see how you could do:
```python3
if reset:
    encoder = make_pipeline(...).encoder.fit(y)
else:
    encoder = self.encoders_[0]
y = encoder.transform(y)
```
But I must be missing something because I don't see how you can get rid of the `else` branch.

Reply:
if it helps, the pseudocode for what I'm trying to achieve here is:
```
if (partial_fit called first time, or self.warm_start or fit called):
    create new transformer
    fit to y
    transform y
else:  # partial_fit called after (fit or partial_fit) or fit called after (fit or partial_fit) & warm_start=True
   grab existing (fitted) transformer
   transform y
save transformer used
return y
```

Or are you referring to the fact that is unpacked from a list and then repacked? That we could change, the pros/const are similar to https://github.com/adriangb/scikeras/pull/88#discussion_r499177021